src/shogun/preprocessor/FisherLDA.cpp

```diff
@@ -250,7 +251,10 @@ bool CFisherLDA::fit(CFeatures *features, CLabels *labels, int32_t num_dimension
 		// x=M
 		// MatrixXd M=Sw.householderQr().solve(Sb);
 		// calculate the eigenvalues and eigenvectors of M.
-		EigenSolver<MatrixXd> es(Sw.householderQr().solve(Sb));
+		SGMatrix<float64_t> M=
+				linalg::qr_solver(SGMatrix<float64_t>(Sw), SGMatrix<float64_t>(Sb));
+		SGMatrix<float64_t>::EigenMatrixXtMap M_eig=M;
+		EigenSolver<MatrixXd> es(M_eig);
 
```

Comment:
@karlnapf So the problem for replacing the solvers (including Cholesky solver) is that the matrices here are already mapped to `Eigen` objects. Now to call `linalg::XX_solver()`, we are casting them back to `SGMatrix`, and most likely they'll be cast again to `Eigen` inside linalg library. Do you think it makes sense? Or should we expect a huge refactoring to limit `Eigen` in linalg library only?
It is going to worth is if we use ViennaCL or XLA backends. But Eigen is the primary one now for sure.